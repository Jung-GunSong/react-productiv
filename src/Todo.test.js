import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const todo = {
  id: 1,
  title: "test1",
  description: "testing1",
  priority: 2,
}

describe("Todo component", function () {
  it("renders without crashing", function () {
    render(<Todo todo={todo} />);
  });

  it("renders as expected by snapshot", function () {
    const {container} = render(<Todo todo={todo} />);

    expect(container).toMatchSnapshot();
  });

  it("renders properly with todo", function () {
    const {getByText} = render(<Todo todo={todo} />);

    expect(getByText("test1")).toBeInTheDocument();
    expect(getByText("testing1")).toBeInTheDocument();
    expect(getByText("priority: 2")).toBeInTheDocument();
  });


});
