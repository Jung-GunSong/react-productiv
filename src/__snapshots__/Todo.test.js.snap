// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders as expected by snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test1
      </b>
       
      <small>
        priority: 
        2
      </small>
    </div>
    <div>
      <small>
        testing1
      </small>
    </div>
  </div>
</div>
`;

exports[`Todo component renders properly with todo 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test1
      </b>
       
      <small>
        priority: 
        2
      </small>
    </div>
    <div>
      <small>
        testing1
      </small>
    </div>
  </div>
</div>
`;
